src/measure_theory/group/geometry_of_numbers.lean

```diff
@@ -0,0 +1,78 @@
+/-
+Copyright (c) 2021 Alex J. Best. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Alex J. Best
+-/
+import analysis.convex.measure
+import measure_theory.group.fundamental_domain
+import measure_theory.measure.haar_lebesgue
+
+/-!
+# Geometry of numbers
+
+In this file we prove some of the fundamental theorems in the geometry of numbers, as studied by
+Hermann Minkowski.
+
+## Main results
+
+* `exists_pair_mem_lattice_not_disjoint_vadd`: Blichfeldt's principle, existence of two distinct
+  points in a subgroup such that the translates of a set by these two points are not disjoint when
+  the covolume of the subgroup is larger than the volume of the
+* `exists_ne_zero_mem_lattice_of_measure_mul_two_pow_lt_measure`: Minkowski's theorem, existence of
+  a non-zero lattice point inside a convex symmetric domain of large enough volume.
+
+## TODO
+
+* Calculate the volume of the fundamental domain of a finite index subgroup
+* Voronoi diagrams
+* See https://arxiv.org/abs/1405.2119 for some more ideas.
+-/
+
+namespace measure_theory
+
+open ennreal finite_dimensional measure_theory measure_theory.measure set
+open_locale pointwise
+
+variables {E L : Type*} [measurable_space E] {μ : measure E} {F s : set E}
```

Comment:
It feels weird to me to have `F` and `s` appear before all the typeclass arguments. Can you put these in the lemma statements instead?

Reply:
"typeclass arguments first" is a bit of a meaningless rule when you get to measure theory where there are many typeclasses some of which depend on non-typeclasses. I certainly won't enforce it here if it leads to duplicating variable declarations.